minio/api.py

```diff
@@ -193,6 +196,14 @@ def make_bucket(self, bucket_name, location='us-east-1'):
         """
         is_valid_bucket_name(bucket_name)
 
+        ## Region already set in constructor, validate if
+        ## caller requested bucket location is same.
+        if self._region:
+            if self._region != location:
+                raise InvalidArgumentError("Configured region {0}, requested"
```

Comment:
This doesnt seem right - how does one make a bucket in a region other than `self._region`? It seems like it is needed to reset the Minio Client object to do this.